Topics/System_Design/Mobile_App_System_Design.md

```diff
@@ -0,0 +1,33 @@
+
+
+
+## Performance 
+
+### 1. Battery Life 
+Mobile devices run on a battery and more often than not will be used in circumstances where the device is not connected to a power source. Therefore mobile apps should be designed to minimize battery usage, which can be achieved by reducing background activity and using power-efficient APIs. If an app causes the battery life to decrease more rapidly it could lead to user dissatisfaction. To expand on the power-efficient APIs, APIs should serve only the bare-minimum content needed for the app to work to limit the amount of information that is parsed and the number of API calls (In the case of a REST API) should be also reduced as much as possible to reduce the amount of processing power needed thus preserving batter life. 
+
+### 2. Resource Constraints 
+The hardware resources available to a mobile device are more limited when compared with a desktop. Therefore it is essential to use resources efficiently to prevent slow response times or app crashes. In general, any computationally expensive task should be outsourced to the backend if the application has one. For example, if an image is served in a mobile app the size of the image should be optimized to accommodate for the resource limitations faced by the mobile device. Another good example is that if an app is implementing a feed or a long list, lazy loading is essential as it will significantly reduce the initial load times of the desired screen.
+
+### 3. Network Variability
+As stated earlier mobile apps are often used on the go, which also means that the network strength may vary. With that in mind, mobile apps should minimize the number of API calls made to the backend. Apps should also cache the data retrieved from a backend API, that is store the retrieved data in a local data store like SQLite on Android or Core Data on iOS. A caching policy should be designed for the application, for example, the app should only make an API call if a particular piece of data is not available in the cache or a sufficient amount of time has passed thus assuming that the data could have changed. Caching will also improve performance, limiting the need for API calls while also making the app less reliant on a strong network.
+
+
+## Conclusion
+In conclusion, the development of a mobile app requires distinct system design decisions, as mobile devices have many aspects that are unique to them. As discussed there are key factors such as user interface design that should be tailored for smaller screens, touch interactions, and contextual use. Additionally, performance considerations should also be included in the system design of a mobile app, in particular battery life, resource constraints, and network variability that can be solved by employing techniques such as caching. When integrated the elements discussed contribute not only to functional mobile apps but also, to engaging and efficiently meeting the needs of their users. 
```

Comment:
The wording could be a little better here: "The development of a mobile app requires system design decisions that are distinct from those on a desktop app."